### PR TITLE
GroundPrimitive

### PR DESCRIPTION
**Pull request by oterral**
Use GroundPrimitive instead of Primitive when altittudeMode is set to clampToGround.

Ready for review @gberaudo .

### CODE REVIEW

File: Cesium.externs.js [hunk 1]
```diff
@@ -1421,6 +1436,23 @@ Cesium.PolylineMaterialAppearance = function(object) {};
...
+ * height: (number|undefined),
+ * extrudedHeight: (number|undefined),
+ * vertexFormat: number
+ * }}
+ */
+Cesium.optionsCorridorGeometry;
+
+/**
+ * @constructor
+ * @param {Object=} object
```
Comment (gberaudo) [inline Cesium.externs.js]: Cesium.optionsCorridorGeometry

File: examples/vectors-clamptoground.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (gberaudo) [inline examples/vectors-clamptoground.html]: I would better not duplicate the vector.js example but enhance it with a button doing:
```js
vectorLayer.set('altitudeMode', 'clampToGround');
vectorLayer2.set('altitudeMode', 'clampToGround');
```

**Comment by gberaudo** [comment]
안녕하세요, @oterral
Nice to see you have started to stab this.

Adding `GroundPrimitive` support and implementing lines on terrain are two different tasks.
The last one needs to be done in Cesium because corridors take meters but we want to use pixels and the conversion depends on the distance. With a fixed ratio, we get the following incorrect reult:
meters_variable_width
pixels_to_meters

See my other comments.

File: src/featureconverter.js [hunk 3]
```diff
@@ -295,21 +309,39 @@ olcs.FeatureConverter.prototype.olLineStringGeometryToCesium =
...
+  var heightReference = this.getHeightReference(layer, feature, olGeometry);
+
+  var outlinePrimitive;
+
+  if (heightReference == Cesium.HeightReference.CLAMP_TO_GROUND) {
+    // Handle both color and width
+    var outlineGeometry = new Cesium.CorridorGeometry({
+      // always update Cesium externs before adding a property
+      positions: positions,
+      width: this.extractLineWidthFromOlStyle(olStyle) * 10000,
```
Comment (gberaudo) [inline src/featureconverter.js]: The conversion factor can not be static, see https://github.com/openlayers/ol3-cesium/pull/243#issuecomment-140387081

File: examples/vectors-clamptoground.html (hunk 2 above)
Comment (oterral) [inline examples/vectors-clamptoground.html]: good idea

**Comment by oterral** [comment]
I agree, using Corridor for outline is not ideal. It's workaround until lines on terrain will be implemented.

For the hardcoded factor.  it will be great if we could do like the scaleByDistance  options of [PointPrimitive](https://cesiumjs.org/Cesium/Build/Documentation/PointPrimitive.html?classFilter=Point)

**Comment by oterral** [comment]
@gberaudo I've removed the use of Corridor geometry

**Comment by gberaudo** [comment]
@oterral, could you please modify the vector example as suggested in https://github.com/openlayers/ol3-cesium/pull/243#discussion_r39507298?

**Comment by oterral** [comment]
Yes I'm doing it.

**Comment by gberaudo** [comment]
@oterral, the clampToGround mode button does not seem to work. Maybe it does not work when geometries have absolute 3D coordinates? If it is the case, you could add a new 2D polygon feature to the vector example? See `examples/data/geojson/vector_data.geojson`.

**Comment by oterral** [comment]
Ok I had not tested I was on windows yesterday and nothing worked

**Comment by oterral** [comment]
Ok it works now, there was a typo issue and the layer must be remove then re-add. Is there a way to r-render a layer without adding/removing it?

**Comment by gberaudo** [comment]
Your workaround is OK. I created an issue
https://github.com/openlayers/ol3-cesium/issues/250.

**Comment by gberaudo** [comment]
It looks good now, please merge.

**Comment by oterral** [comment]
Thx for the review but I can't merge I haven't the right :)

**Comment by gberaudo** [comment]
OK, done

**Pull request merged**